packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -139,16 +139,20 @@ export class Bootstrapper {
     *    - the name indicating the custom permissions boundary to be used
     * Re-bootstrapping will NOT be blocked by either tightening or relaxing the permissions' boundary.
     */
-    const currentPermissionsBoundary = current.parameters.InputPermissionsBoundary;
+    const currentPermissionsBoundary: string | undefined = current.parameters.InputPermissionsBoundary || undefined;
     const inputPolicyName = params.examplePermissionsBoundary ? CDK_BOOTSTRAP_PERMISSIONS_BOUNDARY : params.customPermissionsBoundary;
-    let policyName;
+    let policyName: string | undefined;
     if (inputPolicyName) {
       // If the example policy is not already in place, it must be created.
       const sdk = (await sdkProvider.forEnvironment(environment, Mode.ForWriting)).sdk;
       policyName = await this.getPolicyName(environment, sdk, inputPolicyName, partition, params);
     }
     if (currentPermissionsBoundary !== policyName) {
-      warning(`Switching from ${currentPermissionsBoundary} to ${policyName} as permissions boundary`);
+      if (!currentPermissionsBoundary) {
+        warning(`Adding new permissions boundary ${policyName}`);
+      } else {
+        warning(`Changing permissions boundary from ${currentPermissionsBoundary} to ${policyName}`);
+      }
```

Comment:
If you're going to go this route you need a special message for "removing" as well.